src/AdvancedHMC.jl

```diff
@@ -139,9 +143,21 @@ include("sampler.jl")
 export sample
 
 include("abstractmcmc.jl")
-export DifferentiableDensityModel
 
-include("contrib/ad.jl")
+Hamiltonian(metric::AbstractMetric, ℓ::LogDensityModel) = Hamiltonian(
+    metric,
+    Base.Fix1(LogDensityProblems.logdensity, ℓ.logdensity),
+    Base.Fix1(LogDensityProblems.logdensity_and_gradient, ℓ.logdensity),
```

Comment:
One final suggestion: I guess it could be useful to check here if the gradient is actually defined with the LogDensityProblems API and in case it is missing one could throw an error with an explanation that e.g. one can choose an AD backend by using the other constructor.